pandas/core/indexes/range.py

```diff
@@ -180,9 +186,12 @@ def _constructor(self):
         """ return the class to use for construction """
         return Int64Index
 
-    @cache_readonly
+    @property
     def _data(self):
-        return np.arange(self._start, self._stop, self._step, dtype=np.int64)
+        if self._cached_data is None:
```

Comment:
can you give this a doc-string (e.g. that cached_data is actually an int array and be constructed only if necessary for performance reasons